File: packages/sdk-python/src/eidolondb/resources/artifacts.py
from __future__ import annotations

from typing import Any

from .._client import EidolonDBClient
from .._types import Artifact, CreateArtifactResponse, DeleteArtifactResponse


class ArtifactsResource:
    def __init__(self, client: EidolonDBClient):
        self._client = client

    def create(self, **kwargs: Any) -> CreateArtifactResponse:
        return self._client.request("POST", "/artifacts", body=kwargs)

    def get(self, artifact_id: str) -> Artifact:
        return self._client.request("GET", f"/artifacts/{artifact_id}")

    def delete(self, artifact_id: str) -> DeleteArtifactResponse:
        return self._client.request("DELETE", f"/artifacts/{artifact_id}")

File: packages/sdk-js/src/resources/artifacts.ts
import { EidolonDBClient } from '../client.js';
import type { Artifact, CreateArtifactInput, CreateArtifactResponse, DeleteArtifactResponse } from '../types.js';

export class ArtifactsResource {
  constructor(private readonly client: EidolonDBClient) {}

  /** Create an artifact, optionally auto-processing into memories. */
  create(input: CreateArtifactInput): Promise<CreateArtifactResponse> {
    return this.client.request<CreateArtifactResponse>('POST', '/artifacts', input);
  }

  /** Get an artifact by ID. */
  get(id: string): Promise<Artifact> {
    return this.client.request<Artifact>('GET', `/artifacts/${id}`);
  }

  /** Delete an artifact and related derived memories. */
  delete(id: string): Promise<DeleteArtifactResponse> {
    return this.client.request<DeleteArtifactResponse>('DELETE', `/artifacts/${id}`);
  }
}
